### ISSUE TITLE
Properly handle Windows WCOW containers in `pkg/rootlessutil`.

### ISSUE THREAD

**Issue by aznashwan** [containerd/nerdctl#2115]
### Description

TL;DR: on Windows, `nerdctl` wrongly thinks it's always running as a rootless parent (read background below), so we should try to reach consensus on whether Windows containers should be considered rootless or not.
Q1: Would you consider Windows containers managed by HCS to be "rootless" or not?
Q2: Should it depend on the isolation mode of the Windows container? (i.e. Hyper-V vs. Process containers?)

### Background.

Containerd and nerdctl can run in containers in [rootless mode](https://rootlesscontaine.rs), where both the containers **as well as the runtime** can be run under non-root-users, thus providing an additional layer of isolation.

On Linux, Nerdctl leverages [RootlessKit](https://github.com/rootless-containers/rootlesskit) to run containers in rootless mode, and performs several checks arounds the codebase using `rootless.IsRootless` which leads to some special codepaths such as:
* setting up [`slirp4dns` for rootless children](https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/container_run_network.go#L323-L328).
* setting AppArmor [policies for rootless children](https://github.com/containerd/nerdctl/blob/main/pkg/infoutil/infoutil_linux.go#L57).
* added setup for [port mappings for rootless children](https://github.com/containerd/nerdctl/blob/main/pkg/ocihook/ocihook.go#L288-L329).

[`rootlessutil.IsRootless()`](https://github.com/containerd/nerdctl/blob/main/pkg/rootlessutil/rootlessutil.go#L29) should return `true` if the process is either a rootless parent or child, whose subchecks consist of:
* `IsRootlessParent()` just checks whether [`os.Getuid() != 0`](https://github.com/containerd/nerdctl/blob/main/pkg/rootlessutil/parent.go#L33).
* `IsRootlessChild()` checks that `!IsRootlessParent()` as well as [whether the container is running within a Linux user namespace](https://github.com/containerd/nerdctl/blob/main/pkg/rootlessutil/child.go#L26).

### Current Windows behavior.

Currently, because `rootlessutil.IsRootlessParent()` checks that [`os.Getuid() != 0`](https://github.com/containerd/nerdctl/blob/main/pkg/rootlessutil/parent.go#L33), and [`os.Getuid()` always returns `-1` on Windows](https://pkg.go.dev/os#Getuid), all `IsRootlessParent()` calls will return `true` on Windows, and thus **all `rootlessutil.IsRootless()` calls will also return `true`**.

Fortunately, this does not really affect any important logic, since most of the special logic is only required if `IsRootlessChild()` is true.

### Discussion.

The discussion boils down to the following questions:
* should Windows Process Containers be considered rootless?
    * probably not, since even if the container is run as `ContainerUser` or some other non-admin user, it'll still be managed by HCS, which is a privileged service and thus breaks the "rootless" definition
* should Windows Hyper-V Containers be considered rootless?
    * probably so when considering that it has its own guest OS in the micro-VM it's running in, and so from the PoV of the containerd host, the container and its runtime "look rootless" despite being root-ful within the micro-VM
    * also, probably not, since a case could be made that HCS/Hyper-V themselves are privileged system services. so anything they run cannot be counted as "rootless"

Depending on the consensus, the actual changes to the codebase I'll be making will be very minimal, but it's important that a consensus is reached so things can be planned better moving forward.

### Misc.

FWIW, I've also **tried hardcoding `false` in all `IsRootless*()` calls on Windows and things seemed to mostly work**, with the only notable pain point I could find were some syslog-related tests which were ignored if `IsRootless*() == true`. (the tests set up their own syslog server, and everything was technically fine, just [that the tests run `echo`s](https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/container_run_log_driver_syslog_test.go#L97) which will contain added double quotes `""` around the message due to the Windows `testutil.CommonImage` having `cmd.exe` as the default entrypoint)

I think it's also be worth noting that `rootlessutil.IsRootless*()` only checks whether the **current** `nerdctl` parent process is rootless, and creates rootless containers based on that.
On Windows, however, the question of rootlessness might depend on the `--isolation` flag, but should not theoretically depend on the `nerdctl` process at all (only precondition will be the user running `nerdctl` having access to containerd's named pipe to make calls to it)

### Describe the results you received and expected

Expected `nerdctl` to have explicit knowledge/logic on whether it's running in rootless mode on Windows or not. (currently, `rootlessutil.IsRootlessParent()` is always `true` due to a minor oversight with `os.Getuid()`, see above)

### What version of nerdctl are you using?

Latest `main`.

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

Will happen on any Windows host.

**Comment by aznashwan** [comment]
@jsturtevant @dcantah @helsaawy your opinion and any added insights you may have would be greatly appreciated!

**Comment by kevpar** [comment]
Windows doesn't support any way (to my knowledge) to run containers without being some sort of Administrator/SYSTEM or otherwise elevated role. Even managing a UVM (micro-VM) requires elevated privileges to interact with the Hyper-V stack. So I would expect that nothing on Windows is treated as "rootless".

Is there some reason we would want Windows containers/runtime to be considered rootless?

**Comment by aznashwan** [comment]
> Is there some reason we would want Windows containers/runtime to be considered rootless?

Not at all, it's just that I noticed that `nerdctl` always thought it was running rootless on Windows so far and wanted to have the matter settled.

I'll be making a short PR to fix this in `rootlessutil` and the minor side-effects that has.
